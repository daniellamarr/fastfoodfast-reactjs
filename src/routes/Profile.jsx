/* eslint-disable no-script-url */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Loader from '../components/Loader.jsx';
import { userData } from '../actions/authActions';
import { getAllOrders } from '../actions/orderActions';

/**
 * Profile
 */
export class Profile extends Component {
  /**
   * Initialized functions for component
   * @returns {null} null
   */
  componentDidMount() {
    this.props.userData();
    this.props.getAllOrders();
  }

  /**
   * @returns {JSX} jsx
   */
  render() {
    const userDetails = this.props.auth.user;
    const orders = this.props.orders.orders.map((order) => {
      let meal = '';
      // eslint-disable-next-line arrow-body-style
      meal += order.food.map((foodData) => {
        return `${foodData.orderid} - ${foodData.quantity}\n`;
      });
      return (
        <tr key={order.id}>
          <td>{`Order ${order.id}`}</td>
          <td>
            <a
              className="default-text"
              href="javascript:;"
            >
              {meal}
            </a>
          </td>
          <td>{order.price}</td>
          <td>{order.status}</td>
        </tr>
      );
    });
    return (
      <div>
        <div className="profileLoader">
          { this.props.orders.orderStatus ? <Loader /> : null }
        </div>
        <section id="profile">
          <div className="profile-details">
            <h3 className="userFullName">
              {userDetails ? userDetails.user.name : null}
            </h3>
            <br />
            <h4 className="userEmail">
              {userDetails ? userDetails.user.email : null}
            </h4>
          </div>
          <div className="profile">
            <center>
              <div className="profileImage" />
            </center>
          </div>
        </section>
        <section id="profile-bar" />
        <section id="profile-section">
          <div className="profileOrders">
            <center>
              <h3>Your Orders</h3>
            </center>
            <div className="table">
              <table className="order-table">
                <thead>
                  <tr>
                    <th>S/N</th>
                    <th>Products</th>
                    <th>Price(N)</th>
                    <th>Status</th>
                  </tr>
                </thead>
                <tbody id="userorders">
                  {orders}
                </tbody>
              </table>
            </div>
          </div>
          <div className="clear"></div>
        </section>
        <div className="error hide"></div>
      </div>
    );
  }
}

Profile.propTypes = {
  userData: PropTypes.func.isRequired,
  getAllOrders: PropTypes.func.isRequired,
  auth: PropTypes.shape({
    user: PropTypes.object,
  }),
  orders: PropTypes.shape({
    orders: PropTypes.array,
    orderStatus: PropTypes.bool,
  }),
};

export const mapStateToProps = state => ({
  auth: state.auth,
  orders: state.orders,
});

export default connect(mapStateToProps, { userData, getAllOrders })(Profile);
