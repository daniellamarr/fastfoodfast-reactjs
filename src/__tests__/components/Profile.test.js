/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Profile } from '../../routes/Profile.jsx';

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);
const store = mockStore();

let wrapper;

const props = {
  userData: () => {},
  getAllOrders: () => {},
  auth: {
    user: {
      user: {
        name: 'Test User',
        email: 'Test Email',
      },
    },
  },
  orders: {
    orders: [
      {
        food: [
          {
            orderid: 'Test food',
            quantity: 5,
          },
        ],
      },
    ],
    orderStatus: true,
  },
};

beforeEach(() => {
  wrapper = mount(
    <Provider store={store}>
      <Profile {...props} />
    </Provider>,
  );
});
afterEach(() => {
  wrapper.unmount();
});

describe('Profile Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return food and quantity', () => {
    expect(wrapper.find('Profile')
      .props().orders.orders[0].food
      .map(foodItem => `${foodItem.orderid} - ${foodItem.quantity}`))
      .toEqual(['Test food - 5']);
  });

  it('empty profile loader class when order status is false', () => {
    const props2 = {
      ...props,
      orders: {
        orders: [
          {
            food: [
              {
                orderid: 'Test food',
                quantity: 5,
              },
            ],
          },
        ],
        orderStatus: false,
      },
    };
    wrapper = mount(<Profile {...props2} />);
    expect(wrapper.find('.profileLoader').render().text()).toEqual('');
  });

  it('empty userFullName, userEmail class when userdetails is empty', () => {
    const props2 = {
      ...props,
      auth: {},
    };
    wrapper = mount(<Profile {...props2} />);
    expect(wrapper.find('.userFullName').render().text()).toEqual('');
    expect(wrapper.find('.userEmail').render().text()).toEqual('');
  });
});
