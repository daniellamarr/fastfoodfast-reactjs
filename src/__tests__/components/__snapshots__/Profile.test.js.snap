// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Profile
    auth={
      Object {
        "user": Object {
          "user": Object {
            "email": "Test Email",
            "name": "Test User",
          },
        },
      }
    }
    getAllOrders={[Function]}
    orders={
      Object {
        "orderStatus": true,
        "orders": Array [
          Object {
            "food": Array [
              Object {
                "orderid": "Test food",
                "quantity": 5,
              },
            ],
          },
        ],
      }
    }
    userData={[Function]}
  >
    <div>
      <div
        className="profileLoader"
      >
        <Loader>
          <div
            id="loader"
          >
            <center>
              <img
                alt="loading..."
                src="test-file-stub"
                width="80px"
              />
            </center>
          </div>
        </Loader>
      </div>
      <section
        id="profile"
      >
        <div
          className="profile-details"
        >
          <h3
            className="userFullName"
          >
            Test User
          </h3>
          <br />
          <h4
            className="userEmail"
          >
            Test Email
          </h4>
        </div>
        <div
          className="profile"
        >
          <center>
            <div
              className="profileImage"
            />
          </center>
        </div>
      </section>
      <section
        id="profile-bar"
      />
      <section
        id="profile-section"
      >
        <div
          className="profileOrders"
        >
          <center>
            <h3>
              Your Orders
            </h3>
          </center>
          <div
            className="table"
          >
            <table
              className="order-table"
            >
              <thead>
                <tr>
                  <th>
                    S/N
                  </th>
                  <th>
                    Products
                  </th>
                  <th>
                    Price(N)
                  </th>
                  <th>
                    Status
                  </th>
                </tr>
              </thead>
              <tbody
                id="userorders"
              >
                <tr
                  key="0"
                >
                  <td>
                    Order undefined
                  </td>
                  <td>
                    <a
                      className="default-text"
                      href="javascript:;"
                    >
                      Test food - 5

                    </a>
                  </td>
                  <td />
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="clear"
        />
      </section>
      <div
        className="error hide"
      />
    </div>
  </Profile>
</Provider>
`;
